### PR TITLE
Creates GitHub Actions CI configuration

### LINKED ISSUE #60: Move back to GitHub Actions
[Travis CI's recent billing changes](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing) could affect this project. We moved away from GHA in #31 because our GH paid plan didn't include GHA. Apparently, it does now, so we could use it instead.

A potential inspiration for a modern GHA configuration: https://github.com/scallop/scallop/blob/develop/.github/workflows/ci.yml

We may be able to tackle #2 as a part of this…

### PR DESCRIPTION
**Pull request by colindean**
Fixes #60

### CODE REVIEW

**Comment by colindean** [comment]
I'm pretty sure this is good to go now. The base CI is definitely working as intended and I'll do a release once this is merged to test out the release workflow.

**Comment by colindean** [comment]
I'd love some eyes on this.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            ~/.sbt
+            ~/.ivy2/cache
+            ~/.coursier/cache/v1
+            ~/.cache/coursier/v1
+            ~/AppData/Local/Coursier/Cache/v1
+            ~/Library/Caches/Coursier/v1
+          key: ${{ runner.os }}-sbt-cache-v2-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+
+      - name: Build, test, and package project
+        run: bin/sbt clean compile test package makePom
```
Comment (c-horn) [inline .github/workflows/ci.yaml:39]: Does the cache action run again after the build?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+jobs:
+  build:
+    uses: target/data-validator/.github/workflows/ci.yaml@main
+  deploy:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      # uses sbt-github-packages, see build.sbt
+      - run: bin/sbt publish
```
Comment (c-horn) [inline .github/workflows/release.yaml:19]: `https://maven.pkg.github.com/target/`?

**Review by c-horn — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (colindean) [reply .github/workflows/ci.yaml:39]: It seems like the cache config sets the build _to be cached_. That is, it doesn't build the cache at at the time of invocation. This was confusing to me but it's in [the example](https://github.com/actions/cache#example-workflow) before any "work" is done…

File: .github/workflows/release.yaml (hunk 2 above)
Comment (colindean) [reply .github/workflows/release.yaml:19]: I think that's where it'll go!

**Review by samratmitra-0812 — APPROVED** [review]

**Pull request merged**